src/enclave-agent/src/config.rs

```diff
@@ -0,0 +1,119 @@
+use serde::{Deserialize, Serialize};
+use std::{fs::File, io::Write, path::Path};
+
+const DEFAULT_KEYPROVIDER_CONFIG_PATH: &str = "/etc/ocicrypt_keyprovider_native.conf";
+const DEFAULT_KEYPROVIDER_CONFIG: &str = r#"
+{
+    "key-providers": {
+        "attestation-agent": {
+            "native": "attestation-agent"
+        }
+    }
+}
+"#;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct DecryptConfig {
+    pub key_provider: Option<String>,
+    #[serde(default = "default_security_validate")]
+    pub security_validate: Option<bool>,
+}
+
+fn default_security_validate() -> Option<bool> {
+    Some(true)
+}
+
+impl DecryptConfig {
+    pub fn load_from_file(file_path: &String) -> Result<Self, std::io::Error> {
+        let file = File::open(file_path)?;
+        let config: DecryptConfig = serde_json::from_reader(file)?;
+        Ok(config)
+    }
+}
+
+pub struct KeyproviderConfig {
+    pub config_path: String,
+    pub is_exist: bool,
+}
+
+impl KeyproviderConfig {
+    pub fn default_config() -> Result<Self, std::io::Error> {
+        let path = DEFAULT_KEYPROVIDER_CONFIG_PATH.to_string();
+        let mut kc = Self {
+            config_path: path.clone(),
+            is_exist: Path::new(&path).exists(),
+        };
+
+        if !kc.is_exist {
+            File::create(Path::new(&path))?.write_all(DEFAULT_KEYPROVIDER_CONFIG.as_bytes())?;
+            kc.is_exist = true;
+        }
+        Ok(kc)
+    }
+
+    pub fn export_to_env(&self) {
+        std::env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", self.config_path.clone());
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_load_decrypt_config() {
+        struct ConfigCase {
+            path: String,
+            result_key_provider: Option<String>,
+            result_security_validate: Option<bool>,
+        }
+
+        let config_cases: Vec<ConfigCase> = vec![
+            ConfigCase {
+                path: "test_data/decrypt_config/decrypt_config_normal.conf".to_string(),
+                result_key_provider: Some(
+                    "provider:attestation-agent:eaa_kbc::127.0.0.1:1234".to_string(),
```

Comment:
Could this constant `"provider:attestation-agent:eaa_kbc::127.0.0.1:1234"` be defined as a `const` in this test mod? Same fow below